ideascube/library/tests/test_views.py

```diff
@@ -108,6 +114,26 @@ def test_book_detail_page_does_not_list_specimens_to_non_staff(app, book):
     resp.mustcontain(no=[specimen.barcode])
 
 
+def test_book_detail_page_shows_warning_message_to_staff_if_no_specimens(staffapp, book):
+    resp = staffapp.get(reverse('library:book_detail', kwargs={'pk': book.pk}))
+    resp.mustcontain("Please add a specimen, or the book won&#39;t be available for the users")
+
+
+def test_book_detail_page_does_not_show_warning_message_to_user_if_no_specimens(app, book):
```

Comment:
Test `test_everyone_should_access_book_detail_page` makes me think that it's what we want.

But I agree it could be changed. (For a 403 ? As the book is existing but we don't want to show it to user)

@yohanboniface, what do you think about this ?